docs/frontend-system/architecture/05-extension-overrides.md

```diff
@@ -10,38 +10,107 @@ description: Frontend extension overrides
 
-## Creating a Extension Override
+## Creating an Extension Override
 
-<!--
+The following steps should be followed to override extensions:
 
-How to create an extension override and export it from a package.A
+```tsx
+// plugins/aperture-overrides/src/overrides.ts
+import {
+  createThemeApi,
+  createExtensionOverrides
+} from '@backstage/frontend-plugin-api';
 
- - Example using the default exports
+// Creating a light theme extension;
+const apertureLightTheme = createThemeApi({ … });
 
-You can also create them directly withing the app for local overrides.
+// Creating a light theme extension;
+const apertureDarkTheme = createThemeApi({ … });
 
- - Example of installing an override directly into an app
+// Creating an extension overrides preset
+export createExtensionOverrides({
+  Extensions: [apertureLightTheme, apertureDarkTheme]
+});
 
-Mention that in can still be a good idea to split your overrides out into separate packages in large projects. But it's up to you to decide how to group the extensions into extension overrides.
+// plugins/aperture-overrides/src/index.ts
+export { default } from './overrides';
+```
 
- -->
+For this example, we are creating overrides for the light and dark theme extensions and exporting the overrides from the plugin index file. Now we are able to use the overrides in a Backstage app:
 
-## Overriding Existing Extensions
+```tsx
+// packages/app/src/App.tsx
+import { createApp } from '@backstage/frontend-app-api';
+import apertureOverrides from ‘@backstage/plugin-aperture-overrides’
+
+const app = createApp({
+ features: [ apertureOverrides ],
+});
+
+export default app.createRoot().
+```
+
+If the plugin you want to change is internal to your company or you just want to replace one of the application's core extensions, you can decide to create replacements directly in the application for local replacements. See an example below:
+
+```tsx
+// packages/app/src/themes.ts
+import { createExtensionOverrides } from from '@backstage/frontend-plugin-api';
+
+// Creating the light theme extension;
+export const apertureLightTheme = createThemeApi({ … });
+
+// Creating the light theme extension;
+const apertureDarkTheme = createThemeApi({ … });
 
-<!--
+// Exporting your custom extensions
+export default [apertureLightTheme, apertureDarkTheme];
+
+// packages/app/src/App.tsx
+import { createApp } from '@backstage/frontend-app-api';
+import themes from ‘./themes’
+
+const app = createApp({
+ features: [
+   createExtensionOverrides({
+     extensions: [
+      ...themes,
+     ],
+   }),
+ ],
+});
+
+export default app.createRoot();
+```
+
+Note that it can still be a good idea to split your overrides out into separate packages in large projects. But it's up to you to decide how to group the extensions into extension overrides.
```

Comment:
I think this could be replaced by the opposite: noting that it's recommended to split into a separate package but you can also keep the overrides directly in the app.

Reply:
Replaced ✅